### PR TITLE
refactor: add param files in planning module

### PR DESCRIPTION
**Pull request by kyoichi-sugahara**
## Description

Modify and add parameter files to match launcher repository. This is necessary for retrieving parameters when launching nodes for testing within each package.

## Tests performed

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/config/avoidance/avoidance.param.yaml [hunk 1]
```diff
@@ -6,6 +6,8 @@
       resample_interval_for_output: 4.0                 # [m]
       detection_area_right_expand_dist: 0.0             # [m]
       detection_area_left_expand_dist: 1.0              # [m]
+      drivable_area_right_bound_offset: 0.0             # [m]
+      drivable_area_left_bound_offset: 0.0              # [m]
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/config/avoidance/avoidance.param.yaml:10]: It already exists in the other config file.

https://github.com/autowarefoundation/autoware.universe/blob/4f0769fd3daa561b6bd3642f14a43cf3a6670290/planning/behavior_path_planner/config/drivable_area_expansion.param.yaml#L4-L7

File: planning/behavior_path_planner/config/behavior_path_planner.param.yaml [hunk 2]
```diff
@@ -28,8 +35,59 @@
...
-    expected_front_deceleration_for_abort: -2.0
-    expected_rear_deceleration_for_abort: -2.5
+    expected_front_deceleration_for_abort: -1.0
+    expected_rear_deceleration_for_abort: -2.0
 
     rear_vehicle_reaction_time: 2.0
-    rear_vehicle_safety_time_margin: 2.0
+    rear_vehicle_safety_time_margin: 1.0
+
+    # USE ONLY WHEN THE OPTION COMPILE_WITH_OLD_ARCHITECTURE IS SET TO FALSE.
```
Comment (satoshi-ota) [inline planning/behavior_path_planner/config/behavior_path_planner.param.yaml:44]: [imo] It is better to separate these params to another file like `autoware_launch` config.

https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/scene_module_manager.param.yaml

File: planning/behavior_path_planner/config/avoidance/avoidance.param.yaml (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/config/avoidance/avoidance.param.yaml:10]: Thank you for checking!
But in launcher side same parameters exist. Launcher side should be deleted?

https://github.com/autowarefoundation/autoware_launch/blob/32c4dadd1254c189ccfcca205e246661014d026e/autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/avoidance/avoidance.param.yaml/#L9-L10

Comment (satoshi-ota) [reply planning/behavior_path_planner/config/avoidance/avoidance.param.yaml:10]: Yes, I think we should remove both side parameters.

File: planning/behavior_path_planner/config/behavior_path_planner.param.yaml (hunk 2 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/config/behavior_path_planner.param.yaml]: Deleted parameters.

File: planning/behavior_path_planner/config/avoidance/avoidance.param.yaml (hunk 1 above)
Comment (kyoichi-sugahara) [reply planning/behavior_path_planner/config/avoidance/avoidance.param.yaml]: I understand!
Deleted params in `avoidance.param.yaml`

**Review by takayuki5168 — APPROVED** [review]

File: planning/behavior_path_planner/config/pull_over/pull_over.param.yaml [hunk 3]
```diff
@@ -1,6 +1,7 @@
 /**:
   ros__parameters:
     pull_over:
+      request_length: 100.0
```
Comment (kosuke55) [inline planning/behavior_path_planner/config/pull_over/pull_over.param.yaml:4]: oh, sorry for missing checking.
some parameters are not in the launcher config.
https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/planning/scenario_planning/lane_driving/behavior_planning/behavior_path_planner/pull_over/pull_over.param.yaml

**Pull request merged**